### PR TITLE
migrator: Create an enterprise version

### PR DESCRIPTION
**Pull request by efritz**
We will soon need to import out-of-band migration runners into the migrator. Some of these are implemented in enterprise-only code. This is a pre-step to allow this import in _some_ version of the migrator.

## Test plan

Tested migrator (oss and enterprise versions) runs locally by hand.

### CODE REVIEW

**Comment by efritz** [comment]
@bobheadxi I'm specifically looking at what places in the CI pipelines I might need to add `migrator` to to ensure it builds the enterprise version.

File: enterprise/cmd/migrator/build.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
```
Comment (eseliger) [inline enterprise/cmd/migrator/build.sh:1]: I think you will want to change the docker image path we build in CI so the enterprise version is built and published.

**Review by eseliger — APPROVED** [review]

File: enterprise/cmd/migrator/build.sh (hunk 1 above)
Comment (eseliger) [reply enterprise/cmd/migrator/build.sh:1]: good news, seems like that's automated: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/dev/ci/internal/ci/operations.go?L714%3A4-721%3A7=

**Comment by bobheadxi** [comment]
> @bobheadxi I'm specifically looking at what places in the CI pipelines I might need to add migrator to to ensure it builds the enterprise version.

I think doing what you did here (adding an equivalent to `enterprise`) is sufficient: https://buildkite.com/sourcegraph/sourcegraph/builds/164678#01825b47-f97e-44d8-8175-85a703c116af/116-118

**Pull request merged**